src/NuGet.Core/NuGet.Build.Tasks/GetRestoreProjectReferencesTask.cs

```diff
@@ -59,11 +59,15 @@ public override bool Execute()
             foreach (var project in ProjectReferences)
             {
                 var refOutput = BuildTasksUtility.GetPropertyIfExists(project, "ReferenceOutputAssembly");
+                var copyToOutputProperty = BuildTasksUtility.GetPropertyIfExists(project, "CopyToOutputDirectory");
```

Comment:
Hey @elangelo Thanks for this PR. 

Unfortunately taking this change as is would introduce some unexpected side-effects. 

The references returned by this task would be used by NuGet for the transitive walk of PackageReferences and the build in order to copy the output. 

I think we likely need to differentiate between the 2 scenarios. 

@rainersigwald am I interpreting this correctly? 
We wouldn't want NuGet to use this reference for the transitive PR walk, but the build would need to be told that this is a runtime dependency only?

Reply:
Another question, for deciding which project references to build, do the build tasks ever look at the assets file or just the msbuild info? 
cc @dsplaisted

Reply:
OK I see now that this PR, although it will eventually also copy the packagereferences, is not ideal as it will actually also add all the packagereferences as references that can be used during compile time which is not what we want at all...

Reply:
So while reading a bit more through the code of Nuget it looks like Nuget never did anything with runtime dependencies. Which makes me feel that the issue should have stayed in msbuild.

Reply:
> OK I see now that this PR, although it will eventually also copy the packagereferences, is not ideal as it will actually also add all the packagereferences as references that can be used during compile time which is not what we want at all...

That's correct. 

> So while reading a bit more through the code of Nuget it looks like Nuget never did anything with runtime dependencies. Which makes me think that the issue should have stayed in msbuild.

Both NuGet & msbuild look at the project reference graphs, so maybe both. I can see a solution to this that involves a change in both NuGet & the SDK/msbuild